### ISSUE TITLE
Kona 2024 EU - start and stop charging

### ISSUE THREAD

**Issue by StephaneG38** [Hyundai-Kia-Connect/kia_uvo#872]
Good morning, I own a Kona 2024 and I live in France. 
Can you add the possibility like on the Bluelink application to be able to:
- start and stop charging 
- modify the charging current (60%, 90% or 100%) 
thanks in advance 
Stephane

**Comment by cdnninja** [comment]
Start stop exists today.   Current doesn't however target soc does.

**Comment by StephaneG38** [comment]
Hello @cdnninja 
I don't have these functions on my integration.
![image](https://github.com/Hyundai-Kia-Connect/kia_uvo/assets/38179281/866b4223-61b0-4caa-aace-29b36d1d0f42)

**Comment by cdnninja** [comment]
As per the readme they are services.

**Comment by StephaneG38** [comment]
I can't start or stop charging because there are no controls to do so. Is this normal or is it due to my installation?

**Comment by cdnninja** [comment]
Normal.  It is a service you can call though.

**Comment by StephaneG38** [comment]
How do I call these services?

**Comment by StephaneG38** [comment]
I think I found
![image](https://github.com/Hyundai-Kia-Connect/kia_uvo/assets/38179281/5f85533a-4984-4829-be62-d53986df4bf7)

**Comment by StephaneG38** [comment]
On the other hand, I don't see the possibility of modifying the charging current (60%, 90% or 100%)

**Comment by cdnninja** [comment]
If someone sniffs the traffic from the native app and creates it sure.  I don't have a Kia anymore nor am I in EU.   This is region specific.   

It wouldn't be too hard to add once someone sniffs the data though.

**Comment by StephaneG38** [comment]
I understand.
I don't know how to sniff them but if you have a link to a tutorial I would like to try.

**Comment by StephaneG38** [comment]
I just tested starting and stopping charging but unfortunately neither works.

![image](https://github.com/Hyundai-Kia-Connect/kia_uvo/assets/38179281/30442720-16e9-4867-b04c-fe008169cb47)

**Comment by cdnninja** [comment]
I think it is in the wiki.  The cliff notes are Knox emulator with Charles proxy.  

Android 6 non rooted was best success historically.

**Comment by StephaneG38** [comment]
Oups
sorry I hadn't seen the wiki
I'll have a look and try
Thank you very much @cdnninja

**Comment by cdnninja** [comment]
All good! I'm on my phone so can't easily drop a link.

**Comment by StephaneG38** [comment]
I've just looked at the Wiki, it's not easy!
I don't know if I can do it!
Don't you have any contacts in Europe who own a Kona 2024 and could do it?

**Comment by cdnninja** [comment]
Nope. This is a volunteer based project.   As such someone local would need to come forward to do it.

**Comment by StephaneG38** [comment]
Ok 
I will try 
thank you for your help

**Comment by StephaneG38** [comment]
Hello @cdnninja 
I have a error when i start apk-mitm : 
![image](https://github.com/Hyundai-Kia-Connect/kia_uvo/assets/38179281/a6f3dd10-dc5b-419f-af78-de32d4a2d586)
[decoding.log](https://github.com/Hyundai-Kia-Connect/kia_uvo/files/15277468/decoding.log)
[encoding-aapt.failed.log](https://github.com/Hyundai-Kia-Connect/kia_uvo/files/15277472/encoding-aapt.failed.log)
[encoding-aapt2.failed.log](https://github.com/Hyundai-Kia-Connect/kia_uvo/files/15277473/encoding-aapt2.failed.log)

thank you for your help !

**Comment by StephaneG38** [comment]
I tried with the UVO file from kia and it works:
![image](https://github.com/Hyundai-Kia-Connect/kia_uvo/assets/38179281/2feb2d5e-afde-4377-bc16-f645df0dccad)


The problem is that for the Hyndai KONA I have Bluelink and not UVO!

**Comment by StephaneG38** [comment]
The UVO APK file was wrong
I start again with another one that seems OK

**Comment by cdnninja** [comment]
I'm not familiar with that tool.

**Comment by StephaneG38** [comment]
I've just patched Bluelink XAPK and installed it on a second tel so as not to conflict with mine.
The application doesn't seem to want to start.
I'll continue the process and keep you informed
![image](https://github.com/Hyundai-Kia-Connect/kia_uvo/assets/38179281/a7f602d3-4209-4dab-bc59-ff4261f19368)

**Comment by cdnninja** [comment]
I have heard patching does that which is why the external approach I referenced sometimes works better.

**Comment by StephaneG38** [comment]
Actually it’s not simple. 
What exactly is the external approach that is simpler?
